src/util.c

```diff
@@ -516,6 +516,46 @@ char *tr_strlsize(char *buf, guint64 bytes, size_t buflen)
     return buf;
 }
 
+/* Take a GList of strings and concatenate them.
+ * Opposite of tr_split_to_list */
+gchar *tr_list_concat(const gchar *separator, GList *list)
+{
+    gchar *str_out;
+    gchar **str;
+
+    guint len = g_list_length(list);
+    if (len == 0)
+        return g_strdup("");
+
+    str = g_malloc0(sizeof(gchar *) * (len + 1));
+
+    for (guint i = 0; i < len; i++) {
+        str[i] = json_node_dup_string(g_list_nth_data(list, i));
```

Comment:
Never use `g_list_nth_data()`, it iterates over the entire list every time its called.

Use `for (GList *l = list; l; l = g_list_next())` for example, it keeps the pointer to the current item and is far more efficient.

Reply:
Also you don't need to dupe the string here. Since the string is only used during the lifetime of this function it is safe to use without transfering ownership.